[2.24.0alpha8] FilteredDeckOptions crashes on open

> 2.24.0alpha8 crashes on opening filtered deck options or after naming a new filtered deck

-- Khonkhortisan

Probably the following. API 30

```
Caused by: java.lang.ClassCastException: android.content.res.TypedArray cannot be cast to java.lang.AutoCloseable
	at com.ichi2.preferences.NumberRangePreference.getMinFromAttributes(NumberRangePreference.kt:118)
	at com.ichi2.preferences.NumberRangePreference.<init>(NumberRangePreference.kt:43)
	at com.ichi2.preferences.IncrementerNumberRangePreference.<init>(IncrementerNumberRangePreference.kt:43)
	... 26 more
```

<details>

<summary>Full stack trace</summary>

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.FilteredDeckOptions}: android.view.InflateException: Binary XML file line #31: Error inflating class java.lang.reflect.Constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3502)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3654)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2116)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7720)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:612)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:997)
Caused by: android.view.InflateException: Binary XML file line #31: Error inflating class java.lang.reflect.Constructor
	at android.preference.GenericInflater.createItem(GenericInflater.java:409)
	at android.preference.GenericInflater.createItemFromTag(GenericInflater.java:439)
	at android.preference.GenericInflater.rInflate(GenericInflater.java:490)
	at android.preference.GenericInflater.rInflate(GenericInflater.java:502)
	at android.preference.GenericInflater.inflate(GenericInflater.java:334)
	at android.preference.GenericInflater.inflate(GenericInflater.java:271)
	at android.preference.PreferenceManager.inflateFromResource(PreferenceManager.java:343)
	at android.preference.PreferenceActivity.addPreferencesFromResource(PreferenceActivity.java:1541)
	at com.ichi2.anki.FilteredDeckOptions.addPreferences(FilteredDeckOptions.kt:264)
	at com.ichi2.anki.FilteredDeckOptions.onCreate(FilteredDeckOptions.kt:239)
	at android.app.Activity.performCreate(Activity.java:8000)
	at android.app.Activity.performCreate(Activity.java:7984)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3475)
	... 11 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at android.preference.GenericInflater.createItem(GenericInflater.java:392)
	... 23 more
Caused by: java.lang.ClassCastException: android.content.res.TypedArray cannot be cast to java.lang.AutoCloseable
	at com.ichi2.preferences.NumberRangePreference.getMinFromAttributes(NumberRangePreference.kt:118)
	at com.ichi2.preferences.NumberRangePreference.<init>(NumberRangePreference.kt:43)
	at com.ichi2.preferences.IncrementerNumberRangePreference.<init>(IncrementerNumberRangePreference.kt:43)
	... 26 more
```

</details>

https://ankidroid.org/acra/app/1/bug/388244/report/8d02a928-c4aa-4f49-b719-d38ec143a95b

--- a/AnkiDroid/src/main/java/com/ichi2/preferences/NumberRangePreference.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/preferences/NumberRangePreference.kt
@@ -21,6 +21,7 @@ import android.text.InputType
 import android.util.AttributeSet
 import android.view.View
 import com.ichi2.anki.R
+import com.ichi2.anki.utils.ext.usingStyledAttributes
 import timber.log.Timber
 
 @Suppress(
@@ -115,8 +116,8 @@ open class NumberRangePreference :
      * This method should only be called once from the constructor.
      */
     private fun getMinFromAttributes(attrs: AttributeSet?): Int =
-        context.obtainStyledAttributes(attrs, R.styleable.NumberRangePreference).use {
-            it.getInt(R.styleable.NumberRangePreference_min, 0)
+        context.usingStyledAttributes(attrs, R.styleable.NumberRangePreference) {
+            getInt(R.styleable.NumberRangePreference_min, 0)
         }
 
     /**
@@ -126,8 +127,8 @@ open class NumberRangePreference :
      * This method should only be called once from the constructor.
      */
     private fun getMaxFromAttributes(attrs: AttributeSet?): Int =
-        context.obtainStyledAttributes(attrs, R.styleable.NumberRangePreference).use {
-            it.getInt(R.styleable.NumberRangePreference_max, Int.MAX_VALUE)
+        context.usingStyledAttributes(attrs, R.styleable.NumberRangePreference) {
+            getInt(R.styleable.NumberRangePreference_max, Int.MAX_VALUE)
         }
 
     /**

